json-endpoints/jetpack/class.jetpack-json-api-cron-endpoint.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+// GET /sites/%s/cron
+class Jetpack_JSON_API_Cron_Endpoint extends Jetpack_JSON_API_Endpoint {
+	protected $needed_capabilities = 'manage_options';
+
+	protected function validate_call( $_blog_id, $capability, $check_manage_active = true ) {
+		parent::validate_call( $_blog_id, $capability, false );
+	}
+
+	protected function result() {
+		return array(
+			'cron_array' => _get_cron_array(),
+			'current_timestamp' => time()
+		);
+	}
+
+	protected function sanitize_hook( $hook ) {
+		return preg_replace( '/[^A-Za-z0-9-_]/', '', $hook );
```

Comment:
In my testing I haven't come across uppercase keys but it doesn't meant that and there are not many actions that user uppercase. I would keep things as they are. Since it is a bit more liberal then the current sanitize_key() implementation. Maybe WP needs to add sanitize_action().